test-data/unit/stubgen.test

```diff
@@ -552,14 +602,148 @@ def f(a): ...
 [case testInferOptionalOnlyFunc]
 class A:
     x = None
-    def __init__(self, a=None) -> None:
+    def method(self, a=None):
         self.x = []
 [out]
 from typing import Any, Optional
 
 class A:
-    x = ...  # type: Any
-    def __init__(self, a: Optional[Any] = ...) -> None: ...
+    x: Any
+    def method(self, a: Optional[Any] = ...): ...
+
+[case testAnnotationImportsFrom]
+import foo
+from collection import defaultdict
+x: defaultdict
+
+[out]
+from collection import defaultdict
+
+x: defaultdict
+
+[case testAnnotationImports]
+import foo
+import collection
+x: collection.defaultdict
+
+[out]
+import collection
+
+x: collection.defaultdict
+
+
+[case testAnnotationImports]
+from typing import List
+import collection
+x: List[collection.defaultdict]
+
+[out]
+import collection
+from typing import List
+
+x: List[collection.defaultdict]
+
+
+[case testAnnotationFwRefs]
+x: C
+
+class C:
+    attr: C
+
+y: C
+[out]
+x: 'C'
```

Comment:
This is just a consequence of the fact that ``.pyi`` files are actually never executed, and the only reason of ForwardRef's is to "hide" something from the Python runtime. This is already heavily used in typeshed stubs to simplify them, if you think it is worth mentioning this in PEP 484, then you could make a short PR to ``python/peps`` repo.